tower-http/src/services/fs/serve_dir.rs

```diff
@@ -62,6 +64,48 @@ impl ServeDir {
         self.buf_chunk_size = chunk_size;
         self
     }
+
+    /// Informs the service that it should look for a precompressed (by the gzip algorithm)
+    /// version of _any_ file in the directory with the correct file extension  that can be
+    /// served if the client have the correct corresponding `Accept-Encoding`.
+    /// (i.e `dir/foo.txt.gz` when the served directory is `dir` and the requested file is `foo.txt`)
+    /// If not the uncompressed version will be served instead. Note that this means
+    /// that both the precompressed version(s) and the uncompressed file is expected
+    /// to be present in the same directory.
```

Comment:
```suggestion
    /// Informs the service that it should look for a precompressed gzip
    /// version of _any_ file in the directory.
    ///
    /// If the client has an `Accept-Encoding` header that allows the gzip encoding, 
    /// the file `dir/foo.txt.gz` will be served instead of `dir/foo.txt`.
    /// If the precompressed file is not available, or the client doesn't support it,
    /// the uncompressed version will be served instead. 
    /// Both the precompressed version and the uncompressed version are expected
    /// to be present in the directory.
```

(ditto similar suggestions to the sibling methods)

I'm not sure where it should be mentioned about the `q` settings for browser preference.